src/common/mfu_daos.c

```diff
@@ -2575,14 +2623,22 @@ static int serialize_dkeys(struct hdf5_args *hdf5,
 
         d_iov_set(&dkey_iov, dkey_enum_buf, ENUM_DESC_BUF);
 
-        /* get dkeys */
-        rc = daos_obj_list_dkey(*oh, DAOS_TX_NONE, &dkey_number,
-                                dkey_kds, &dkey_sgl, &dkey_anchor,
-                                NULL);
-        if (rc != 0) {
-            MFU_LOG(MFU_LOG_ERR, "failed to list dkeys: %d", rc);
-            goto out;
+        if (ofeat & DAOS_OF_KV_FLAT) {
+            rc = daos_kv_list(*oh, DAOS_TX_NONE, &dkey_number,
+                              dkey_kds, &dkey_sgl, &dkey_anchor, NULL);
+            if (rc != 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to list dkeys: %d", rc);
+                goto out;
+            }
+        } else {
+            rc = daos_obj_list_dkey(*oh, DAOS_TX_NONE, &dkey_number,
+                                    dkey_kds, &dkey_sgl, &dkey_anchor, NULL);
+            if (rc != 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to list dkeys: %d", rc);
```

Comment:
@dsikich formatted daos error
```suggestion
                MFU_LOG(MFU_LOG_ERR, "failed to list dkeys: "DF_RC, DP_RC(rc));
```